Add new synchronization permissions

ArvindHarinder1 opened the pull request.

FaithOmbongi commented on changelog/Microsoft.DirectoryServices.json:
```diff
@@ -1,5 +1,31 @@
+          "ApiChange": "Permission",
+          "ChangedApiName": "Synchronization.ReadWrite.All",
+          "ChangeType": "Addition",
+          "Description": "Added the `Synchronization.ReadWrite.All` application and permission to configure synchronization settings.",
+          "Target": "synchronization"
+        },
+        {
+          "Id": "00c006a9-0eaf-423f-9ed1-ge86808382ab",
+          "ApiChange": "Permission",
+          "ChangedApiName": "Synchronization.Read.All",
+          "ChangeType": "Addition",
+          "Description": "Added the `Synchronization.Read.All` application and permission to read synchronization settings.",
```
Suggestion
```suggestion
          "Description": "Added the `Synchronization.ReadWrite.All` application and delegated permissions to configure [synchronization settings](https://learn.microsoft.com/en-us/graph/api/resources/synchronization-overview?view=graph-rest-beta).",
          "Target": "synchronization"
        },
        {
          "Id": "00c006a9-0eaf-423f-9ed1-ge86808382ab",
          "ApiChange": "Permission",
          "ChangedApiName": "Synchronization.Read.All",
          "ChangeType": "Addition",
          "Description": "Added the `Synchronization.Read.All` application and delegated permissions to read [synchronization settings](https://learn.microsoft.com/en-us/graph/api/resources/synchronization-overview?view=graph-rest-beta).",
```

FaithOmbongi reviewed (changes requested):
Hi @ArvindHarinder1 -

In the spirit of encouraging least privilege, we're avoiding documenting _Directory.Read.All_ and _Directory.ReadWrite.All_ if an API doesn't **need** either. If _Synchronization.Read.All_ and _Synchronization.ReadWrite.All_ are enough for the synchronization operations, please remove the Directory.* permissions from the permissions tables. Let me know if we need to document either.

Let one suggestion for the changelog.

ArvindHarinder1 commented:
> Hi @ArvindHarinder1 -
> 
> * In the spirit of encouraging least privilege, we're avoiding documenting _Directory.Read.All_ and _Directory.ReadWrite.All_ if an API doesn't **need** either. If _Synchronization.Read.All_ and _Synchronization.ReadWrite.All_ are enough for the synchronization operations, please remove the Directory.* permissions from the permissions tables. Let me know if we need to document either.
> * Also, document the new permissions in the [permissions reference](https://learn.microsoft.com/en-us/graph/permissions-reference).
> * Left one suggestion for the changelog.

@FaithOmbongi  thanks for the feedback. I've gone ahead and addressed the feedback.

FaithOmbongi commented:
Selected the re-request review icon.

FaithOmbongi reviewed (approved):
Thank you @ArvindHarinder1 for your work on this PR. Approved.
